CardScan/Classes/ScanBaseViewController.swift

```diff
@@ -197,17 +201,23 @@ import Vision
     override open func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         
-        self.videoFeed.setup(captureDelegate: self) { success in
-            if success {
-                self.setupMask()
-            }
+        if self.ocr.numbers.count > 0 && self.ocr.expiries.count > 0 {
+            self.ocr.numbers.removeAll()
+            self.ocr.expiries.removeAll()
+            self.ocr.firstResult = nil
         }
-        
+        self.calledOnScannedCard = false
+        //MARK: ^Jaime just try the remove all
         self.videoFeed.willAppear()
         self.isNavigationBarHidden = self.navigationController?.isNavigationBarHidden ?? true
         self.navigationController?.setNavigationBarHidden(true, animated: animated)
     }
     
+    open override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(true)
+        self.setupMask()
```

Comment:
I'd love to see a comments + pointer to apple docs about why this is here.

Reply:
setUp mask is not a related to AVFoundation, it is our UI logic. What should i make a pointer to?